lib/src/list.dart

```diff
@@ -66,64 +73,73 @@ class RealmList<T extends Object> extends collection.ListBase<T> {
     }
   }
 
-  /// Sets the element at the specified index in the list.
   @override
   void operator []=(int index, T value) {
     RealmListInternal.setValue(handle, realm, index, value);
   }
 
-  /// Clears the collection in memory and the references
-  /// to the objects in this collection in Realm.
+  @override
+  void clear() => realmCore.listClear(this);
 
-  /// Removes all elements from this list.
-  ///
-  /// The length of the list becomes zero.
-  /// If the elements are managed [RealmObject]s, they all remain in the Realm.
   @override
-  void clear() {
-    realmCore.listClear(this);
+  bool get isValid => realmCore.listIsValid(this);
+}
+
+class UnmanagedRealmList<T extends Object> extends collection.ListBase<T> with RealmEntity implements RealmList<T> {
+  late final _unmanaged = <T?>[]; // lazy ctor'ed (use T? for length=)
```

Comment:
```suggestion
  late final _unmanaged = <T?>[];
```

Reply:
I guess the comment is outdated, no?

Reply:
also being lazy here brings no value since we initialize it every time in the ctor.

Reply:
Actually we don't if the optional `items` is null, but it is a very minor optimisation.

Reply:
The fact we use `T?` not T, is a requirement to use `ListBase`.

Reply:
dropped `late` and changed comment